### PR TITLE
Fixes #13889,#13991 - Enable modifications to red hat repositories

### PR DESCRIPTION
**Pull request by jlsherrill**
this adds the ability to update some fields on redhat repos, such as download policy
and mirror_on_sync.  In addition this changes away the feed certs and ca are specified
to attempt to be more straight forward

### CODE REVIEW

**Comment by jlsherrill** [comment]
TODO:  

- [ ] More testing
- [ ] Write tests
- [ ] fix more ui references to product.readonly

**Comment by jlsherrill** [comment]
@johnpmitsch if you wanna look over this too, that'd be great, it touches a bit of the code you worked on

**Comment by johnpmitsch** [comment]
@jlsherrill just double checking, [these values](https://github.com/Katello/katello/blob/master/app/models/katello/candlepin/content.rb#L67-L69) don't have to be changed, right?

**Comment by johnpmitsch** [comment]
@jlsherrill so creating, syncing, and updating a repo works for both custom and redhat repos, and I was able to sync after updating a RH repo which I couldn't before.

I'm having issues syncing a RH repo to a capsule. I am not sure if this falls under the jurisdiction of this PR, but I am seeing

```Katello::Errors::PulpError: RPM1004: Error retrieving metadata: Not found```

Talking to @daviddavis  about this, it looks like maybe the red hat cdn certs aren't being used when a repo is created on a capsule [here](https://github.com/Katello/katello/blob/master/app/lib/actions/katello/capsule_content/create_or_update.rb#L33-L35)?

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 1]
```diff
@@ -338,8 +337,11 @@ def find_gpg_key
     end
 
     def repository_params
-      keys = [:url, :gpg_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name, :download_policy, :mirror_on_sync, :ostree_branches => []]
+      keys = [:download_policy, :mirror_on_sync, :ostree_branches => []]
       keys += [:label, :content_type] if params[:action] == "create"
+      if params[:action] == 'create' || @repository.custom?
+        keys += [:url, :gpg_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name, ]
```
Comment (ehelms) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Trailing comma

File: app/controllers/katello/api/v2/repositories_controller.rb [hunk 2]
```diff
@@ -338,8 +337,11 @@ def find_gpg_key
     end
 
     def repository_params
-      keys = [:url, :gpg_key_id, :unprotected, :name, :checksum_type, :docker_upstream_name, :download_policy, :mirror_on_sync, :ostree_branches => []]
+      keys = [:download_policy, :mirror_on_sync, :ostree_branches => []]
```
Comment (ehelms) [inline app/controllers/katello/api/v2/repositories_controller.rb]: Guessing that ostree_branches parameter might clash with some of the ostree update work

File: app/models/katello/glue/pulp/repos.rb [hunk 3]
```diff
@@ -238,18 +238,7 @@ def repo_url(content_url)
...
-          ca = nil
-          cert = nil
-          if repo.environment.library? && repo.content_view.default?
-            key = self.key
-            cert = self.certificate
-            ca = upstream_ca
-          end
-          repo.refresh_pulp_repo(ca, cert, key)
-        end
+        repos.each { |repo| repo.refresh_pulp_repo }
```
Comment (ehelms) [inline app/models/katello/glue/pulp/repos.rb]: Could reduce to `repos.each(&:refresh_pulp_repo)`

File: app/views/katello/api/v2/products/show.json.rabl [hunk 4]
```diff
@@ -34,8 +34,21 @@ node(:published_content_view_ids) do |product|
...
 
-node :readonly do |product|
+node :redhat do |product|
   product.redhat?
 end
 
+#deprecated
+node :readonly do |product|
+  product.editable?
+end
```
Comment (ehelms) [inline app/views/katello/api/v2/products/show.json.rabl]: Technically we are bumping major versions so you could drop `readonly`

File: app/views/katello/api/v2/products/show.json.rabl [hunk 5]
```diff
@@ -34,8 +34,21 @@ node(:published_content_view_ids) do |product|
...
+
+node :permissions do |product|
+  {
+      :destroy_products => product.editable?,
+      :edit_products => product.deletable?,
+      :sync_products => product.syncable?
```
Comment (ehelms) [inline app/views/katello/api/v2/products/show.json.rabl]: nitpick: this looks over indented

File: app/controllers/katello/api/v2/repositories_controller.rb (hunk 2 above)
Comment (jlsherrill) [inline app/controllers/katello/api/v2/repositories_controller.rb]: it did! and now its deleted

**Comment by thomasmckay** [comment]
ACK - works great

**Comment by jlsherrill** [comment]
Ended up having to rerecord some of the cassettes I believe due to things 'just working' when they were nil before with disregard to what they should be.

**Comment by thomasmckay** [comment]
re-ACK

**Pull request merged**